### PR TITLE
Markers should be properly downcasted and upcasted on bogus paragraphs

### LINKED ISSUE #13285: Markers are not correctly downcasted when start before document list element
## 📝 Provide detailed reproduction steps (if any)

1.  Create a document list that includes only "simple" items (only text).
2.  Indent one of the items in track changes mode.
3.  Call `editor.getData()`.
4.  There is no attribute/HTML element for suggestion start.
5.  When such data will be saved and loaded, the suggestion is gone (as it is incorrect markup).

https://user-images.githubusercontent.com/1502228/212714545-d7bdbd6b-8d04-4913-a413-40ddfb661934.mov

### PR DESCRIPTION
**Pull request by niegowski**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Fix (engine, list, table): Markers should not be lost on list items and in table cells in the data pipeline. Closes #13285.

---

### Additional information

### CODE REVIEW

File: packages/ckeditor5-engine/src/view/domconverter.ts [hunk 1]
```diff
@@ -558,7 +559,8 @@ export default class DomConverter {
 			}
 
 			const transparentRendering = childView.is( 'element' ) &&
-				( childView.getCustomProperty( 'dataPipeline:transparentRendering' ) as boolean | undefined );
+				( childView.getCustomProperty( 'dataPipeline:transparentRendering' ) as boolean | undefined ) &&
```
Comment (arkflpc) [inline packages/ckeditor5-engine/src/view/domconverter.ts:562]:
```suggestion
				!!childView.getCustomProperty( 'dataPipeline:transparentRendering' ) &&
```

Comment (arkflpc) [reply packages/ckeditor5-engine/src/view/domconverter.ts:562]: Or leave it `unknown`. It's used in boolean context only - it doesn't matter what's inside.

**Comment by Mgsy** [comment]
Not a big deal, but I can't reproduce it on `release`:

### Steps to reproduce

1.  Go to http://localhost:8125/ckeditor5/external/collaboration-features/packages/ckeditor5-collaboration/tests/manual/all-features-rtc-document-list.html.
2.  Insert an empty table and put the selection inside an empty cell.
3.  Turn on Track Changes.
4.  Select _Heading 1_ from the headings dropdown to add a formatting suggestion.
5.  Call `editor.data.set( editor.getData(), { suppressErrorInCollaboration: true } )`.
6.  Reject the suggestion.
7.  Try to create an ordered list using autoformatting in the cell that had a formatting suggestion.

### Actual result

It's impossible to insert a list using autoformatting.

**Comment by scofalik** [comment]
AFAICS, this happens because there is a bogus `nbsp;` added to the table cell.

`editor.getData()` returns this:

```
<figure class="table"><table><tbody><tr><td><p data-suggestion-start-before="formatBlock:698dn3bun9mf:e321952b3e500000ea901c56954d7ddcd:e168fe479815a10c23f0c08a44996fe42"><suggestion-end name="formatBlock:698dn3bun9mf:e321952b3e500000ea901c56954d7ddcd:e168fe479815a10c23f0c08a44996fe42"></suggestion-end>&nbsp;</p></td></tr></tbody></table></figure>
```

@niegowski, can we do anything relatively simple to avoid these spaces?

**Comment by niegowski** [comment]
> AFAICS, this happens because there is a bogus `nbsp;` added to the table cell.
> 
> `editor.getData()` returns this:
> 
> ```
> <figure class="table"><table><tbody><tr><td><p data-suggestion-start-before="formatBlock:698dn3bun9mf:e321952b3e500000ea901c56954d7ddcd:e168fe479815a10c23f0c08a44996fe42"><suggestion-end name="formatBlock:698dn3bun9mf:e321952b3e500000ea901c56954d7ddcd:e168fe479815a10c23f0c08a44996fe42"></suggestion-end>&nbsp;</p></td></tr></tbody></table></figure>
> ```
> 
> @niegowski, can we do anything relatively simple to avoid these spaces?

This is a block filler that was not stripped because of another element in the same block (there is a marker-end element) and `DomConverter` is not able to strip this block filler because it does not know whether the marker element is a marker or any other element. We might teach `DomConverter` what element names are describing markers (by registering element names while registering a marker converter) to ignore such elements while checking block filler but I'm not sure if we could do it in such a short time window. This looks like sth out of the scope of this ticket and I would suggest extracting this to a separate issue. WDYT @scofalik

**Comment by scofalik** [comment]
I thought so.

I briefly look whether this won't mess up with revision history, but it seems it is fine.

**Pull request merged**